src/qibo_client/qibo_job.py

```diff
@@ -0,0 +1,221 @@
+import tarfile
+import tempfile
+import time
+import typing as T
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import Path
+
+import qibo
+import requests
+
+from . import constants
+from .config_logging import logger
+from .utils import QiboApiRequest
+
+
+def convert_str_to_job_status(status: str):
+    return next((s for s in QiboJobStatus if s.value == status), None)
+
+
+class QiboJobStatus(Enum):
+    QUEUED = "to_do"
+    RUNNING = "in_progress"
+    DONE = "success"
+    ERROR = "error"
+
+
+def wait_for_response_to_get_request(
+    url: str, seconds_between_checks: T.Optional[int] = None, verbose: bool = False
+) -> T.Tuple[requests.Response, QiboJobStatus]:
+    """Wait until the server completes the computation and return the response.
+
+    :param url: the endpoint to make the request
+    :type url: str
+
+    :return: the response of the get request
+    :rtype: requests.Response
+    :return: the completed job response status
+    :rtype: QiboJobStatus
+    """
+    if seconds_between_checks is None:
+        seconds_between_checks = constants.SECONDS_BETWEEN_CHECKS
+    if not verbose:
+        logger.info("Please wait until your job is completed...")
```

Comment:
This message should appear only if the job is not done. Could you please add a check?

Reply:
If we want to log this message only once, I am forced to add another api request outside the while loop to ask for the job status. The code will be really ugly in my opinion. Do you have an alternative implementation?

Btw in case the job is done the the messages displayed by the function will immediately be the following:

````
Please wait until your job is completed...
Job COMPLETED
```

Which is fine as well in my opinion

Reply:
@marcorossi5 if you can fix this, I will merge.

Reply:
If you want to log it once the price is the following code. Which makes a repeated API call outside the while loop. Are you ok with this? Personally I don't like it

```python
def wait_for_response_to_get_request(
    url: str, seconds_between_checks: T.Optional[int] = None, verbose: bool = False
) -> T.Tuple[requests.Response, QiboJobStatus]:
    if seconds_between_checks is None:
        seconds_between_checks = constants.SECONDS_BETWEEN_CHECKS
    response = QiboApiRequest.get(url, timeout=constants.TIMEOUT)
        job_status = convert_str_to_job_status(response.headers["Job-Status"])
    if not verbose and job_status not in in [QiboJobStatus.DONE, QiboJobStatus.ERROR]:
        logger.info("Please wait until your job is completed...")

    while True:
        response = QiboApiRequest.get(url, timeout=constants.TIMEOUT)
        job_status = convert_str_to_job_status(response.headers["Job-Status"])
        if verbose and job_status == QiboJobStatus.QUEUED:
            logger.info("Job QUEUING")
        if verbose and job_status == QiboJobStatus.RUNNING:
            logger.info("Job RUNNING")
        if job_status in [QiboJobStatus.DONE, QiboJobStatus.ERROR]:
            if verbose:
                logger.info("Job COMPLETED")
            return response, job_status
        time.sleep(seconds_between_checks)
```

Reply:
Ok, any other alternative?

Reply:
An exotic alternative that defines an adaptor for the logger that allows the following syntax `logger.info("message", only_once=True)`. But it seems an overkill and probably not that maintenable.

Otherwise, leave it as it is is not an option?
In the case you call the function with a completed job you'll get the following response.

```
Please wait until your job is completed...
Job COMPLETED
```

Which is not that bad I guess. What is the downside of this?

Reply:
Ok, please go ahead with the second option.

Reply:
You mean this? `logger.info("message", only_once=True)`

Reply:
No, I mean adding the "Job COMPLETED" tag after the please wait, so the user gets more information.

Reply:
we can then remove the `if verbose` line just before `logger.info("Job COMPLETED")` so that this information is always printed even if the user didn't ask